gcloud-java-dns/src/test/java/com/google/gcloud/dns/testing/LocalDnsHelperTest.java

```diff
@@ -0,0 +1,1429 @@
+    record = records.get(0);
+    assertNull(record.getName());
+    assertNull(record.getRrdatas());
+    assertNotNull(record.getType());
+    assertNull(record.getTtl());
+    assertNull(resourceRecordSetListResult.pageToken());
+    options.put(DnsRpc.Option.FIELDS, "nextPageToken");
+    resourceRecordSetListResult = RPC.listDnsRecords(ZONE1.getName(), options);
+    records = ImmutableList.copyOf(resourceRecordSetListResult.results());
+    record = records.get(0);
+    assertNull(record.getName());
+    assertNull(record.getRrdatas());
+    assertNull(record.getType());
+    assertNull(record.getTtl());
+    assertNull(resourceRecordSetListResult.pageToken());
+    options.put(DnsRpc.Option.FIELDS, "nextPageToken,rrsets(name,rrdatas)");
+    options.put(DnsRpc.Option.PAGE_SIZE, 1);
+    resourceRecordSetListResult = RPC.listDnsRecords(ZONE1.getName(), options);
+    records = ImmutableList.copyOf(resourceRecordSetListResult.results());
+    assertEquals(1, records.size());
+    record = records.get(0);
+    assertNotNull(record.getName());
+    assertNotNull(record.getRrdatas());
+    assertNull(record.getType());
+    assertNull(record.getTtl());
+    assertNotNull(resourceRecordSetListResult.pageToken());
+  }
+
+  @Test
+  public void testListChanges() {
+    // no such zone exists
+    try {
+      RPC.listChangeRequests(ZONE_NAME1, EMPTY_RPC_OPTIONS);
+      fail();
+    } catch (DnsException ex) {
+      // expected
+      assertEquals(404, ex.code());
+      assertTrue(ex.getMessage().contains("managedZone"));
+    }
+    // zone exists but has no changes
+    RPC.create(ZONE1, EMPTY_RPC_OPTIONS);
+    Iterable<Change> results = RPC.listChangeRequests(ZONE1.getName(), EMPTY_RPC_OPTIONS).results();
+    ImmutableList<Change> changes = ImmutableList.copyOf(results);
+    assertEquals(0, changes.size());
+    // zone has changes
+    RPC.applyChangeRequest(ZONE1.getName(), CHANGE1, EMPTY_RPC_OPTIONS);
+    RPC.applyChangeRequest(ZONE1.getName(), CHANGE2, EMPTY_RPC_OPTIONS);
+    RPC.applyChangeRequest(ZONE1.getName(), CHANGE_KEEP, EMPTY_RPC_OPTIONS);
+    results = RPC.listChangeRequests(ZONE1.getName(), EMPTY_RPC_OPTIONS).results();
+    changes = ImmutableList.copyOf(results);
+    assertEquals(3, changes.size());
+    // error in options
+    Map<DnsRpc.Option, Object> options = new HashMap<>();
+    options.put(DnsRpc.Option.PAGE_SIZE, 0);
+    try {
+      RPC.listChangeRequests(ZONE1.getName(), options);
+      fail();
+    } catch (DnsException ex) {
+      // expected
+      assertEquals(400, ex.code());
+      assertTrue(ex.getMessage().contains("parameters.maxResults"));
+    }
+    options.put(DnsRpc.Option.PAGE_SIZE, -1);
+    try {
+      RPC.listChangeRequests(ZONE1.getName(), options);
+      fail();
+    } catch (DnsException ex) {
+      // expected
+      assertEquals(400, ex.code());
+      assertTrue(ex.getMessage().contains("parameters.maxResults"));
+    }
+    options.put(DnsRpc.Option.PAGE_SIZE, 15);
```

Comment:
Actually, you test this below in testListPaging, so not necessary.  You do test page size of 1 in the other "testList<Class>" methods, but I think you can remove those b/c they are duplicated in the "test<Class>Paging" methods.